microsoft_outlook_single_tenant/models/microsoft_outlook_mixin.py

```diff
@@ -0,0 +1,18 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import models
+
+
+class MicrosoftOutlookMixin(models.AbstractModel):
+    _inherit = "microsoft.outlook.mixin"
+
+    @property
+    def _OUTLOOK_ENDPOINT(self):
+        outlook_endpoint = "https://login.microsoftonline.com/{path}/oauth2/v2.0/"
+        path = (
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param("microsoft_outlook_single_tenant.tenant_id", "common")
+        )
+        return outlook_endpoint.format(path=path)
```

Comment:
```suggestion
from odoo.addons.microsoft_outlook.models.microsoft_outlook_mixin import MicrosoftOutlookMixin as MOM


class MicrosoftOutlookMixin(models.AbstractModel):
    _inherit = "microsoft.outlook.mixin"

    @property
    def _OUTLOOK_ENDPOINT(self):
        tenant_name = (
            self.env["ir.config_parameter"]
            .sudo()
            .get_param("microsoft_outlook_single_tenant.tenant_id", "common")
        )
        return MOM._OUTLOOK_ENDPOINT.replace(
            "/common/",
            f"/{tenant_name}/"
        )
```

Unblocking little suggestion to avoid replacing initial URL, if the class approach works on this case.

Reply:
MOM? And why using replace + `f` string? :thinking: 

Just use `super()._OUTLOOK_ENDPOINT` to get the original... but I thought we wanted to take full control over the URL and not rely on what odoo uses. Otherwise, let's not replace anything and define our own url only for this case.